xla/service/gpu/gpu_latency_hiding_scheduler.cc

```diff
@@ -141,6 +162,70 @@ CanonicalAsyncOp GpuGetCanonicalAsyncOp(const HloInstruction& hlo) {
   }
 }
 
+bool GpuScheduleCrossesOverlapLimit(
+    const DefaultSchedulerCore::SchedulingState& sched_state,
+    const HloGraphNode* node) {
+  for (const auto& [resource, limit] : sched_state.max_concurrent_resource) {
+    // No resources in flight of this kind. Continue.
+    auto it = sched_state.resource_occupiers_in_flight.find(resource);
+    if (it == sched_state.resource_occupiers_in_flight.end() ||
+        it->second.size() == 0) {
+      continue;
+    }
+    // Number of instances of 'resource' needed if this instruction was
+    // to be scheduled.
+    const int64_t num_resources_needed =
+        sched_state.async_tracker->GetNumResourcesPerInstruction(
+            resource, node->GetInstr());
+    if (limit < num_resources_needed) {
+      return true;
+    }
+  }
+
+  if (node->GetResources().size() == 0) {
+    return false;
+  }
+  auto resource_type = node->GetResources().at(0).first;
+  // If the candidate collective has more than 1 overlapping ranks with
+  // in-flight collectives, they can form cyclic dependency and cannot be
+  // overlapped
+  if ((resource_type - AsyncTracker::GetFirstTargetDefinedResource()) ==
+          static_cast<int64_t>(GpuResourceType::kGpuAsyncStreamCollectives) &&
+      sched_state.resource_occupiers_in_flight.contains(resource_type) &&
+      sched_state.resource_occupiers_in_flight.at(resource_type).size() > 0) {
+    const HloInstruction& curr_hlo_inst = node->GetInstr();
+    if (hlo_query::IsAsyncCollectiveDoneOp(&curr_hlo_inst, true)) {
```

Comment:
Since we use `async_tracker` to figure out whether it's a start or done op, let's make it consistent and use it here too.

Please change to `async_tracker_->IsSupportedAsyncDone` and to `async_tracker_->IsSupportedAsyncStart` below.